ansible/roles/test/tasks/bgp_speaker.yml

```diff
@@ -0,0 +1,153 @@
+  failed_when: False
+
+- name: Store the value of VLAN IPs
+  set_fact:
+     vlan_ips={{ generated_ips}}
+
+- debug: msg="{{generated_ips}}"
+
+- name: Generate two ips in bgp speaker peer range
+  get_ip_in_range: num=2 prefix={{ minigraph_bgp_peers_with_range[0]['ip_range'] }}
+  become: no
+  connection: local
+
+- name: Set the value of ips in bgp speaker peer range
+  set_fact: slb_ips={{generated_ips}}
+
+- debug: msg="{{ generated_ips }}"
+
+- name: Create directory /exabgp to store exabgp related files.
+  file: path=/root/exabgp state=directory
+  connection: local
+
+- name: set exabgp folder name
+  set_fact:
+     exabgp_dir="/root/exabgp"
+
+- name: Create directory /exabgp to store exabgp related files.
+  file: path=/root/exabgp state=directory
+  delegate_to: "{{ptf_host}}"
+
+- name: Generate configuration for the first exabgp instance
+  template: src=roles/test/templates/exabgp/config.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: "config_1.ini", local_ip: '{{slb_ips[0]}}', port_num: '5000'}
+  connection: local
+
+- name: Generate configuration for the second exabgp instance
+  template: src=roles/test/templates/exabgp/config.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: "config_2.ini", local_ip: '{{slb_ips[1]}}', port_num: '6000'}
+  connection: local
+
+- name: Generate configuration for the third exabgp instance
+  template: src=roles/test/templates/exabgp/config.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: "config_3.ini", local_ip: '{{vlan_ips[0]}}', port_num: '7000'}
+  connection: local
+
+- name: Set the prefix to be announced
+  set_fact: announce_prefix="10.10.10.10/26"
+
+- name: Generate routes to be announced
+  template: src=roles/test/templates/exabgp/routes.j2 dest=/tmp/exabgp/routes
+  with_items:
+    - {slb_ip: '{{vlan_ips[0]}}', mux_ip_1: '{{vlan_ips[1]}}', mux_ip_2: '{{vlan_ips[2]}}', port_num_1: '5000', port_num_2: '6000', port_num_3: '7000'}
+  connection: local
+
+- name: Generate start file for exabgp instances
+  template: src=roles/test/templates/exabgp/start.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: 'start.sh', config_file_1: 'config_1.ini', config_file_2: 'config_2.ini', config_file_3: 'config_3.ini', phy_ip: '{{vlan_ips[0]}}', logical_ip_1: '{{slb_ips[0]}}', logical_ip_2: '{{slb_ips[1]}}'}
+  connection: local
+
+- name: Copy exabgp related files to ptf container
+  copy: src=/tmp/exabgp/ dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/dump.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/http_api.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/announce_routes.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
```

Comment:
you can copy the the whole directory to the dest using copy module. do not need to specify file one-by-one.